File: frontend/src/shared/models/ExerciseDto.ts
export type ExerciseType = {
	createdAt: string
	exercise: {
		description: string
		id: string
		name: string
		video?: string
	}

	reps: number
	sets: number
	updatedAt: string
};

export class ExerciseDto {
	public id: string = '';
	public name: string = '';
	public description: string = '';
	public video: string | null = null;
	public createdAt: string = '';
	public updatedAt: string = '';
	public sets: number = 0;
	public reps: number = 0;

	constructor(base: ExerciseType) {
		this.id = base.exercise.id;
		this.name = base.exercise.name;
		this.description = base.exercise.description;
		this.video = base.exercise.video ?? null;
		this.sets = base.sets;
		this.reps = base.reps;
	}
}

File: backend/Coacher.Backend.Contracts/Dto/ExerciseDto.cs
namespace Coacher.Backend.Contracts.Dto;

public class ExerciseDto
{
    public Guid Id { get; set; }
    public string Name { get; set; } = string.Empty;
    public string Description { get; set; } = String.Empty;
    public string? Video { get; set; }
    public int? ExerciseType { get; set; } = 0;
}